processor/deltatocumulativeprocessor/processor.go

```diff
@@ -0,0 +1,76 @@
+package deltatocumulativeprocessor // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor"
+
+import (
+	"context"
+	"errors"
+
+	"go.uber.org/zap"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/processor"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor/internal/data"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor/internal/delta"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor/internal/metrics"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor/internal/streams"
+)
+
+var _ processor.Metrics = (*Processor)(nil)
+
+type Processor struct {
+	next consumer.Metrics
+
+	log    *zap.Logger
+	ctx    context.Context
+	cancel context.CancelFunc
+
+	nums streams.Aggregator[data.Number]
+}
+
+func newProcessor(cfg *Config, log *zap.Logger, next consumer.Metrics) *Processor {
+	ctx, cancel := context.WithCancel(context.Background())
+
+	proc := Processor{
+		log:    log,
+		ctx:    ctx,
+		cancel: cancel,
+		next:   next,
+
+		nums: delta.Numbers(),
+	}
+
+	return &proc
+}
+
+func (p *Processor) Start(ctx context.Context, host component.Host) error {
+	return nil
+}
+
+func (p *Processor) Shutdown(ctx context.Context) error {
+	p.cancel()
+	return nil
+}
+
+func (p *Processor) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{MutatesData: true}
+}
+
+func (p *Processor) ConsumeMetrics(ctx context.Context, md pmetric.Metrics) error {
+	var errs error
+
+	metrics.Each(md, func(m metrics.Metric) {
+		switch m.Type() {
+		case pmetric.MetricTypeSum:
+			sum := m.Sum()
+			if sum.AggregationTemporality() == pmetric.AggregationTemporalityDelta {
+				err := streams.Update(metrics.Sum(m), p.nums)
+				errs = errors.Join(errs, err)
```

Comment:
You accumulate errors, but it doesn't look like you do anything with them. I'm not very familiar with the `ConsumeMetrics()` interface, but after metrics.Each(), should you check `if errs != nil`?

Reply:
great catch!